extensions/pkg/terraformer/terraformer.go

```diff
@@ -255,6 +248,8 @@ func (t *terraformer) execute(ctx context.Context, command string) error {
 
 			if terminationMessage != "" {
 				podLogger.V(1).Info("Termination message of Terraformer pod: " + terminationMessage)
+			} else if ctx.Err() != nil {
+				podLogger.V(1).Info("Context error: " + ctx.Err().Error())
```

Comment:
Should we still try to read the logs of the pod and fill the `terminationMessage`? It might be helpful to see what the Terraformer has done in case something goes wrong and is analyzed later on.

Reply:
It can't be done with the `ctx` in this case, so a new context will have to be constructed, similarly to how it's done for `Delete`. I considered this but didn't implement it this way for the following reason: if the context is in error (cancelled or deadline exceeded), then the reason for aborting the operation is external to the terraformer and can't be found in the terraformer pod logs. The logs will be empty or "normal" and truncated at some point.

Also, I recently became aware that logs with `V(1)` are not logged by extensions by default and there is no mechanism to enable this. Even if we introduce such a mechanism in the future, to see `V(1)` logs one would need to explicitly enable them and reproduce the situation that caused the original error.

Reply:
> then the reason for aborting the operation is external to the terraformer and can't be found in the terraformer pod logs. The logs will be empty or "normal" and truncated at some point.

My motivation was that the logs of the Terraform execution might be helpful in case we have to analyze issues or bugs later on - independent of the reason why the context errored. WDYT?

> Also, I recently became aware that logs with V(1) are not logged by extensions by default and there is no mechanism to enable this. Even if we introduce such a mechanism in the future, to see V(1) logs one would need to explicitly enable them and reproduce the situation that caused the original error.

True, probably the `V(1)` should be eliminated entirely then?

Reply:
> My motivation was that the logs of the Terraform execution might be helpful in case we have to analyze issues or bugs later on - independent of the reason why the context errored. WDYT?

Perhaps, but at least this is not the intention in the current code. The logs are already only read if the termination message is empty. Clearly the intention is to provide a clue why the pod execution failed, not context for additional investigation.

> True, probably the V(1) should be eliminated entirely then?

Why, if this never came up as an issue so far? I think we should limit the verbosity of `Info` logs to what's really useful, and add a mechanism to increase the level to `Debug` when really needed. After discussing this with @timebertt I also included some `V(1)` logs in the watchdogs implementation with the same motivation.

In short, I still think this PR should focus on what was originally intended (making sure that the Terraformed pod is properly cleaned up if the context was cancelled), and not on changing what logs are displayed in case of failure and with what logging level.

I would also like to follow up with issues to introduce a parameter for increasing the log level for at least the provider extensions.